test/end2end/app_test.dart

```diff
@@ -72,9 +95,177 @@ void main() {
     await createAccountToHome(tester);
     await homeToProfilePage(tester);
     await bottomNavigation(tester);
-    await createPost(tester);
-    await deletePost(tester);
+    await createPost(tester, testPostTitle, testPostDescription);
+    await deletePost(tester, testPostTitle, testPostDescription);
   });
+
+  testWidgets("Challenge creation and completion", (WidgetTester tester) async {
+    // create a post that will be the challenge
+    final otherUser = await FirestoreUserGenerator.addUser(fakeFireStore);
+    final postLocation = GeoPointGenerator.createOnEdgeInsidePosition(
+      startLocation,
+      ChallengeRepositoryService.maxChallengeRadius,
+    );
+    final post =
+        FirestorePostGenerator().createUserPost(otherUser.uid, postLocation);
```

Comment:
Good jov with this test :+1: 
I just think we could instanciate the `FirestorePostGenerator` in the setup like that we don't need to re-instanciate it in each test.
What do you think ?

Reply:
1df14d23b2b0e155c512a111bfc91c5b3f36c958